### PR TITLE
Modernize for .NET 6.0 LTS

### PR DESCRIPTION
**Pull request by yaakov-h**
Essentially #1002 but properly.

### CODE REVIEW

**Comment by yaakov-h** [comment]
After this is merged I'll do a separate PR that deals with Trimming.

File: .github/workflows/steamkit2-build.yaml [hunk 1]
```diff
@@ -48,7 +48,11 @@ jobs:
 
     - name: Run Tests
       if: matrix.configuration == 'Debug'
-      run: dotnet test --verbosity normal SteamKit2/Tests/Tests.csproj
+      run: dotnet test --verbosity normal --framework net6.0 SteamKit2/Tests/Tests.csproj
+
+    - name: Run Tests (.NET Framework)
+      if: matrix.configuration == 'Debug' && startsWith(matrix.os, 'windows-')
+      run: dotnet test --verbosity normal --framework net6.0 SteamKit2/Tests/Tests.csproj
```
Comment (xPaw) [inline .github/workflows/steamkit2-build.yaml:55]: wrong framework?

File: SteamKit2/SteamKit2/Util/Win32Helpers.cs [hunk 2]
```diff
@@ -25,123 +29,126 @@ static class Win32Helpers
 
 		static string GetBootDiskLogicalVolume()
 		{
-			return Environment.GetEnvironmentVariable( "SystemDrive" );
+			var volume = Environment.GetEnvironmentVariable( "SystemDrive" );
```
Comment (xPaw) [inline SteamKit2/SteamKit2/Util/Win32Helpers.cs:32]: tabs

File: SteamKit2/Tests/AsyncJobFacts.cs [hunk 3]
```diff
@@ -396,16 +396,27 @@ public async Task AsyncJobMultipleContinuesAsynchronously()
...
             asyncJob.AddResult( new Callback { JobID = 123 } );
 
-            await continuation;
+            WaitForTaskWithoutRunningInline( continuation );
 
             Assert.NotEqual( -1, continuationThreadID );
             Assert.NotEqual( completionThreadID, continuationThreadID );
         }
+
+        static void WaitForTaskWithoutRunningInline( Task task )
```
Comment (xPaw) [inline SteamKit2/Tests/AsyncJobFacts.cs:411]: Add `[MethodImpl(MethodImplOptions.NoInlining)]`?

**Review by xPaw — APPROVED** [review]

File: SteamKit2/Tests/AsyncJobFacts.cs (hunk 3 above)
Comment (yaakov-h) [reply SteamKit2/Tests/AsyncJobFacts.cs:411]: This isn't about compiler inlining but about whether the runtime runs the task "inline" in the wait function - reaching `TaskScheduler.TryRunInline`

**Pull request merged**